core/src/main/java/org/mskcc/cbio/portal/scripts/ImportCosmicData.java

```diff
@@ -65,18 +77,22 @@ public void importData() throws IOException, DaoException {
                 }
                 
                 String id = parts[2];
-                if (!id.matches("COSM[0-9]+")) {
+                if (!id.matches("COS(M|V)[0-9]+")) {
                     System.err.println("Wrong cosmic ID: "+id);
                 } else {
                     id = id.substring(4);
                 }
                 
-                Matcher m = p.matcher(parts[7]);
+                String infoColumnValue = parts[7];
+                Matcher m = p.matcher(infoColumnValue);
                 if (m.find()) {
-                    String gene = m.group(1);
-//                    if (gene.contains("_ENST")) {
-//                        gene = gene.substring(0,gene.indexOf("_ENST"));
-//                    }
+                    Map<String, String> fieldValues = evaluateFieldValues(infoColumnValue);
+
+                    String gene = fieldValues.get(geneEntryName);
+                           
+                    if (gene.contains("_ENST")) {
+                        gene = gene.substring(0,gene.indexOf("_ENST"));
+                    }
```

Comment:
I noticed that we comment this part out currently, could you tell me why do we need to add this back?

Reply:
Hi @dippindots, thank you for reviewing the PR!

I have uncommented it because the new cosmic files have genes with that prefix (e.g. `GENE=PLCH2_ENST00000378488`), and this code allows to insert the record instead of getting the message `Gene symbol in COSMIC not recognized: PLCH2_ENST00000378488`.